mapsource.cc

```diff
@@ -134,12 +142,9 @@ mps_wpt_q_deinit(queue* whichQueue)
  *
  */
 static Waypoint*
-mps_find_wpt_q_by_name(const queue* whichQueue, const QString& name)
+mps_find_wpt_q_by_name(const QList<Waypoint *>* whichQueue, const QString& name)
 {
-  queue* elem, *tmp;
-
-  QUEUE_FOR_EACH(whichQueue, elem, tmp) {
-    Waypoint* waypointp = reinterpret_cast<Waypoint*>(elem);
+  foreach (Waypoint* waypointp, *whichQueue) {
```

Comment:
This could turn into indexOf.

Or we could move mps and gdb into deprecate and never look at this code again. :-/